tests/gocase/unit/keyspace_test.go

```diff
@@ -0,0 +1,206 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package unit
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/apache/incubator-kvrocks/tests/gocase/util"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeyspace(t *testing.T) {
+	srv := util.StartServer(t, map[string]string{})
+	defer srv.Close()
+
+	ctx := context.Background()
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	t.Run("DEL against a single item", func(t *testing.T) {
+		key := "x"
+		value := "foo"
+		require.NoError(t, rdb.Set(ctx, key, value, 0).Err())
+		require.Equal(t, value, rdb.Get(ctx, key).Val())
+		require.Equal(t, int64(1), rdb.Del(ctx, key).Val())
+		require.Equal(t, "", rdb.Get(ctx, key).Val())
+	})
+
+	t.Run("Vararg DEL", func(t *testing.T) {
+		require.NoError(t, rdb.Set(ctx, "foo1", "a", 0).Err())
+		require.NoError(t, rdb.Set(ctx, "foo2", "b", 0).Err())
+		require.NoError(t, rdb.Set(ctx, "foo3", "c", 0).Err())
+		require.Equal(t, int64(3), rdb.Del(ctx, "foo1", "foo2", "foo3").Val())
+		require.Equal(t, []interface{}{nil, nil, nil}, rdb.MGet(ctx, "foo1", "foo2", "foo3").Val())
+	})
+
+	t.Run("KEYS with pattern", func(t *testing.T) {
+		for _, key := range []string{"key_x", "key_y", "key_z", "foo_a", "foo_b", "foo_c"} {
+			require.NoError(t, rdb.Set(ctx, key, "hello", 0).Err())
+		}
+		keys := rdb.Keys(ctx, "foo*").Val()
+		sort.Slice(keys, func(i, j int) bool {
+			return keys[i] < keys[j]
+		})
+		require.Equal(t, []string{"foo_a", "foo_b", "foo_c"}, keys)
+	})
+
+	t.Run("KEYS to get all keys", func(t *testing.T) {
+		keys := rdb.Keys(ctx, "*").Val()
+		sort.Slice(keys, func(i, j int) bool {
+			return keys[i] < keys[j]
+		})
+		require.Equal(t, []string{"foo_a", "foo_b", "foo_c", "key_x", "key_y", "key_z"}, keys)
+	})
+
+	t.Run("DEL all keys", func(t *testing.T) {
+		vals := rdb.Keys(ctx, "*").Val()
+		require.Equal(t, int64(len(vals)), rdb.Del(ctx, vals...).Val())
+		require.NoError(t, rdb.Do(ctx, "dbsize", "scan").Err())
+		time.Sleep(100 * time.Millisecond)
+		require.Equal(t, int64(0), rdb.Do(ctx, "dbsize").Val())
+	})
+
+	t.Run("EXISTS", func(t *testing.T) {
+		newKey := "newkey"
+		require.NoError(t, rdb.Set(ctx, newKey, "test", 0).Err())
+		require.Equal(t, int64(1), rdb.Exists(ctx, newKey).Val())
+		require.Equal(t, int64(1), rdb.Del(ctx, newKey).Val())
+		require.Equal(t, int64(0), rdb.Exists(ctx, newKey).Val())
```

Comment:
If you use `require.EqualValues`, you don't need to cast between integers. But this is no a requirement. Just for your information and current style is OK for me, too.

Reply:
This's good suggestion, have done.